### PR TITLE
netusb: allocate packets from RX pool for incoming frames

### PR DESCRIPTION
**Pull request by xodus7**
Signed-off-by: Corey Wharton <xodus7@cwharton.com>

### CODE REVIEW

**Review by jfischer-no — APPROVED** [review]

File: subsys/usb/device/class/netusb/function_ecm.c [hunk 1]
```diff
@@ -297,15 +297,15 @@ static void ecm_read_cb(uint8_t ep, int size, void *priv)
 		}
 	}
 
-	pkt = net_pkt_alloc_with_buffer(netusb_net_iface(), size,
-					AF_UNSPEC, 0, K_FOREVER);
+	pkt = net_pkt_rx_alloc_with_buffer(netusb_net_iface(), size, AF_UNSPEC,
+					   0, K_FOREVER);
 	if (!pkt) {
-		LOG_ERR("no memory for network packet\n");
+		LOG_ERR("no memory for network packet");
```
Comment (mniestroj) [inline subsys/usb/device/class/netusb/function_ecm.c:303]: In general such changes (removal of newlines) deserve a separate commit.

**Review by mniestroj — APPROVED** [review]
